Source/CocoaAsyncSocket.h

```diff
@@ -6,7 +6,12 @@
 //  CocoaAsyncSocket project is in the public domain.
 //
 
+#if __has_feature(modules)
+#if __has_warning("-Watimport-in-framework-header")
+#pragma clang diagnostic ignored "-Watimport-in-framework-header"
+#endif
 @import Foundation;
+#endif
```

Comment:
```suggestion
#import <Foundation/Foundation.h>
```

Echoing what @robbiehanson said, I think the correct fix is to just not use `@import`